### PR TITLE
provider: add development provider

### PR DESCRIPTION
**Pull request by marquinhoalm**
- Add development provider to handle test requests
- Change test/create folder architecture

closes pagarme/superbowleto#213

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel confortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

**Comment by marquinhoalm** [comment]
@grvcoelho o @Thor99 fez as alterações e vamos utilizar valores de `amount` para controlar os `status` para provider `development`

File: src/providers/development/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    .tap((response) => {
+      logger.info({
+        status: 'succeeded',
+        metadata: {
+          status: response.status,
+          data: response.data
+        }
+      })
+    })
+    .catch((err) => {
```
Comment (mccraveiro) [inline src/providers/development/index.ts]: dá para usar `tapCatch` aqui :)

File: test/functional/boleto/create/bradesco/pending.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ test.before(async () => {
   await purgeQueue(BoletosToRegisterQueue)
 })
 
-test.after(async () => {
+test.after(() => {
```
Comment (mccraveiro) [inline test/functional/boleto/create/bradesco/pending.js]: Não tem problema tirar o async aqui? Ele não deveria retornar a `restoreFunction`?

Comment (marquinhoalm) [inline test/functional/boleto/create/bradesco/pending.js]: tirei o async já que não existe um `await` dentro deste after

Comment (Thor99) [inline test/functional/boleto/create/bradesco/pending.js]: @mccraveiro :)

**Pull request merged**